frontend/src/components/learner/browser/CourseCard.tsx

```diff
@@ -12,23 +12,24 @@ import { ReactComponent as DocumentIcon } from "../../../assets/document.svg";
 import { DEFAULT_IMAGE } from "../../../constants/DummyData";
 import RouterLink from "../../common/RouterLink";
 import { COURSE_OVERVIEW_BASE_ROUTE } from "../../../constants/Routes";
-import { makeDeferredCallback } from "../../../utils/CallbackUtils";
 
 const CourseCard = (): React.ReactElement => {
   const [focused, setFocused] = useState<boolean>(false);
   const [hovered, setHovered] = useState<boolean>(false);
-  const onFocusChange = makeDeferredCallback(setFocused);
-  const onHoverChange = makeDeferredCallback(setHovered);
+  const handleFocus = () => setFocused(true);
+  const handleBlur = () => setFocused(false);
+  const handleMouseEnter = () => setHovered(true);
+  const handleMouseLeave = () => setHovered(false);
```

Comment:
Yeah, this would be more similar to what I had before. In the vein of "explicit > implicit" though, I like how the existing callbacks have explicit names.